modules/dxp/apps/saml/saml-opensaml-integration/src/main/java/com/liferay/saml/opensaml/integration/internal/servlet/profile/WebSsoProfileImpl.java

```diff
@@ -1046,21 +1046,13 @@ private void _bindSamlSsoRequestContext(
 				return;
 			}
 
-			samlSsoRequestContexts = new LinkedHashMap<>();
+			samlSsoRequestContexts = new LinkedHashMap<>(
+				_samlConfiguration.getMaxSamlSsoRequestContexts());
```

Comment:
Hey @ChrisKian ,

It seems that initializing the HashMap with a capacity will not limit its size, only define an initial capacity to it. So it will resize whenever it reaches the initial capacity to accommodate more entries. 

But, while doing some investigation on this, I've found that LinkedHashMap actually has a protected method which [removes the eldest entry](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/LinkedHashMap.html#removeEldestEntry(java.util.Map.Entry)) and I ended up discovering that we have an implementation of this at [LRUMap.java](https://github.com/liferay-appsec/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/util/LRUMap.java)

What do you think about we using `LRUMap` instead?

Reply:
That should work fine @manuelecastro, nice find!